prow/statusreconciler/controller_test.go

```diff
@@ -832,6 +893,9 @@ func TestControllerReconcile(t *testing.T) {
 					if actual, expected := fpjt.created, map[prKey]sets.String{}; !reflect.DeepEqual(actual, expected) {
 						t.Errorf("did not create expected ProwJob: %s", diff.ObjectReflectDiff(actual, expected))
 					}
+					if actual, expected := fpjt.skipped, map[prKey]sets.String{}; !reflect.DeepEqual(actual, expected) {
+						t.Errorf("did not skip expected ProwJob: %s", diff.ObjectReflectDiff(actual, expected))
+					}
```

Comment:
This is a lot of copy and paste. Why does the checker function need to be defined per test case? It looks like it only varies in the hard-coded expected values (which could just be parameterized as fields in the test case struct).